pycbc/libutils.py

```diff
@@ -31,7 +31,12 @@
 # library rather than first going to the global symbol table
 # This should be used when multiple libraries may export incompatible
 # symbols
-RTLD_DEEPBIND = 0x00008
+#
+# This is only defined for linux systems, on macosx, the default behavior
+# is similar
+RTLD_DEEPBIND = 0
+if hasattr(os, 'RTLD_DEEPBIND'):
+    RTLD_DEEPBIND = os.RTLD_DEEPBIND
 
```

Comment:
so maybe taking @spxiwh 's comment into account, what we want here is:
```
if hasattr(os, 'RTLD_DEEPBIND'):
    DEFAULT_RTLD_MODE = os.RTLD_DEEPBIND
else:
   DEFAULT_RTLD_MODE = ctypes.DEFAULT_MODE
```
and then change below to:
```
def get_ctypes_library(libname, packages, mode=DEFAULT_RTLD_MODE)
```
which then also allows override in individual libraries.  Or would you rather export something like `LIBUTILS_RTLD_DEEPCOPY` and have that the default, so that we are then also explicitly setting that in `fftw` rather than relying on the default?  Although it is arguably also confusing to have something that implies you're doing `DEEPCOPY` but in fact on some os's it is a noop.

Reply:
@josh-willis Yeah, either way, non-perfect, but I think I'll opt for just setting up the default as you've proposed and fftw won't set the mode at all.

Reply:
LGTM assuming all unit tests pass. And I agree we should do a release after this.